modules/core/keeper/msg_server.go

```diff
@@ -68,7 +68,6 @@ func (k *Keeper) RegisterCounterparty(goCtx context.Context, msg *clientv2types.
 	// initialize next sequence send to enable packet flow
 	k.ChannelKeeperV2.SetNextSequenceSend(ctx, msg.ClientId, 1)
 
-	k.ClientKeeper.DeleteClientCreator(ctx, msg.ClientId)
```

Comment:
Isn't this a problem since the counterparty info can now change at anytime? Which is not just a DOS but also a security concern

Reply:
I think the goal was to add that as a separate message later, but maybe the way to deal with this is to disable sending on a client until the client creator has been deleted (once we add that)?

Reply:
Hmm yes, i forgot this was the main protection against rerunning this message as well. I will instead ensure that you cannot register if it has already been set.

Reply:
Another option might be to either:
1. Don't allow sending on clients that still has a creator set
2. Delete client creator here still, such that setting the client params can only be done between client creation and setting the counterparty.

Reply:
No I want the creator here to be set indefinitely because I want the ability for creators to change the allowed relayer list long after it has been initially set up